### DISCUSSION TITLE
Custom build - how to set module parameters?

### DISCUSSION THREAD

**Discussion by dali-dbr** [arendst/Tasmota#21769] [Q&A]
Hi.

I managed to find most of the settings in my_user_config.h file. However, I can not find a way to set module parameters like:¸
![image](https://github.com/user-attachments/assets/c785e080-31f4-448e-a9f1-7257468da086)

Also, is there a way to rename those switches for example Switch1 to WindowSensor1 right in custom build, like switchtext later in a console?

Thank you
Br

**Comment by sfromis** [comment]
The way to do that is to **not** be using the configure module setup, but instead a template. You can create that template in the Tasmota web UI, or if you already have the gpios in "Configure Module", you can create a template from that using the command `Template 0`. See the example in `my_user_config.h` for setting `USER_TEMPLATE`. To activate it too, you may want a `#define MODULE USER_MODULE`.

BTW. The intended structure for doing a custom build is to leave `my_user_config.h` alone, and instead make the modification via `user_config_override.h`, which is easier to migrate forward with new Tasmota source versions.

You can embed various commands (including `SwitchText`) in `#define USER_BACKLOG`.

**Comment by dali-dbr** [comment]
Thank you for your prompt reply.

My Template 0 output is the same as the initial one (from [this page](https://templates.blakadder.com/wireless_tag_WT32-ETH01.html))  that I entered before I assigned Swithces and numbers to GPIOS.

![image](https://github.com/user-attachments/assets/513cb77e-0b05-4749-971b-b070a457c2a3)

When I applied this configuration all GPIOS were set to None. After, I needed to assign the switches and numbers to GPIOS using the configure module setup. That's my question. Can I predefine it in a custom build?

This is my current my_user_config.h just for an info. At that point I'm just playing with GITPOD. Didn't yet try to flash the board.

```
#ifdef MODULE
#undef MODULE
#endif
#define MODULE                 USER_MODULE   // [Module] Select default model (the list is kModuleNiceList() in file tasmota_template.h) USER_MODULE is the TEMPLATE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE        USER_MODULE   // to Select the default model as FALLBACK when the user does a RESET 1 

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE          "{\"NAME\":\"WT32_ETH01\",\"GPIO\":[1,1,1,1,1,1,0,0,1,0,1,1,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,0,1,1,0,0,1],\"FLAG\":0,\"BASE\":1}"
```
I will use user_config_override.h as you suggest.

What I'm trying to do is to have a final build that will reset to this exact state when I reset (Button1 for 40 seconds). 

Thank you

Br

**Reply by sfromis** [reply]
Looks like you omitted this point about the template:
> To activate it too, you may want a `#define MODULE USER_MODULE`.

This means that when Tasmota initializes first time, it will be with the template active. I'd not really recommend also setting `FALLBACK_MODULE` to the same, as this will prevent the Tasmota recovery feature from working in case of something in the template resulting in a crash.

Having that `#define` is equivalent to issuing the command `Module 0`, or selecting the template, the entry with with `(0)`, in "Configure Module".

This is also what happens when you follow the guide "How to use Templates?" (link in the left column near the bottom), including this important point:
> Make sure you check Activate

**Comment by dali-dbr** [comment]
You got me a bit lost, but I'm doing my best trying to understand.

> This means that when Tasmota initializes first time, it will be with the template active. 

But that's exactly what I need. To make the board work from the first time.

My goal is to include everything I need in the custom build and have the board running at first boot. If it's possible?

Thank you

**Reply by sfromis** [reply]
Yes, and you were already on the right track.

In addition, for context I mentioned other ways to activate a template, since what you posted about not getting gpio configs, even with a template, looks like you had not followed the linked guide about how to use templates, to ensure that it was also activated.

And maybe reevaluate if it really was such a good idea that you also let `FALLBACK_MOdULE` activate the same template instead of actually having a more basic fallback.

**Reply by dali-dbr** [reply]
Thank you for all the info.

I will try this tomorrow and see how it works. Probably I will have some ore questions later on.

Thank you

Br,

**Comment by dali-dbr** [comment]
So to completely understand the basics about....

- I never modify my_user_config.h. I just leave it as it is.
- I copy and modify the lines I need from  **my_user_config.h** to **user_config_override.h** and even if duplicate, the system will use the ones inside **user_config_override.h**

Then I would need to execute console commands somehow. You said USER_BACKLOG.

//#define USER_BACKLOG
#define Module 0
#define SwitchText1 DW_Sensor1
.
.
.
#define SwitchText8 DW_Sensor8

Ans somehow to assign Switches to GPIOS for the selected configuration.
![image](https://github.com/user-attachments/assets/30c349db-1d75-474b-99c1-c3f3a00b6343)

Thank you

**Reply by sfromis** [reply]
No, I suggested to use the `SwitchText` **command**, not invent non-existing `#define` symbols. `USER_BACKLOG` simply has a string (wrapped in `"`) with a list of commands in the same way as you do with the `Backlog` command in the Tasmota console

And instead of "somehow" assigning switches, use a combined template, as already discussed.

When it comes to duplicate `#define` symbols, this is not automatic, but the docs page for a custom build already describes you how can do overrides.
https://tasmota.github.io/docs/Compile-your-build/#changing-default-settings

**Reply by dali-dbr** [reply]
Thank you for your patience!!! I know it must be pain for you to deal with users not knowing the basics :)

So this template does it all for me
{"NAME":"WT32_ETH01","GPIO":[1,1,1,1,1,1,0,0,1,0,1,1,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,0,1,1,0,0,1],"FLAG":0,"BASE":1,"CMND":"Module 0|GPIO2 160|GPIO4 161|GPIO12 162|GPIO14 163|GPIO15 164|GPIO35 165|GPIO36 166|GPIO39 167|switchtext1 DW1|switchtext2 DW2|switchtext3 DW3|switchtext4 DW4|switchtext5 DW5|switchtext6 DW6|switchtext7 DW7|switchtext8 DW8"}

It will assign GPIOS and switch text for all the switches.

I guess that is the correct use of the template?

Thank you
Br

**Comment by dali-dbr** [comment]
I'm guessing you were suggesting this:

#define USER_BACKLOG "Module 0|GPIO2 160|GPIO4 161|GPIO12 162|GPIO14 163|GPIO15 164|GPIO35 165|GPIO36 166|GPIO39 167|switchtext1 DW1|switchtext2 DW2|switchtext3 DW3|switchtext4 DW4|switchtext5 DW5|switchtext6 DW6|switchtext7 DW7|switchtext8 DW8"}

but semicolon as a separator so:
#define USER_BACKLOG "Module 0;GPIO2 160;GPIO4 161;GPIO12 162;GPIO14 163;GPIO15 164;GPIO35 165;GPIO36 166;GPIO39 167;switchtext1 DW1;switchtext2 DW2;switchtext3 DW3;switchtext4 DW4;switchtext5 DW5;switchtext6 DW6;switchtext7 DW7;switchtext8 DW8"}

```
#define PROJECT                "Dali_DW"         // PROJECT is used as the default topic delimiter

// If not selected the default will be SONOFF_BASIC
#ifdef MODULE
#undef MODULE
#endif
#define MODULE                 USER_MODULE   // [Module] Select default model (the list is kModuleNiceList() in file tasmota_template.h) USER_MODULE is the TEMPLATE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE        WT32_ETH01   // to Select the default model as FALLBACK when the user does a RESET 1 

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE          "{\"NAME\":\"WT32_ETH01\",\"GPIO\":[1,1,1,1,1,1,0,0,1,0,1,1,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,0,1,1,0,0,1],\"FLAG\":0,\"BASE\":1}" // [Template] Set JSON template


#define SAVE_DATA              1                 // [SaveData] Save changed parameters to Flash (0 = disable, 1 - 3600 seconds)
#define SAVE_STATE             true              // [SetOption0] Save changed power state to Flash (false = disable, true = enable)
#define BOOT_LOOP_OFFSET       1                 // [SetOption36] Number of boot loops before starting restoring defaults (0 = disable, 1..200 = boot loops offset)


#define MQTT_SWITCHES          true             // [SetOption114] Detach switches from relays and send MQTT messages instead
#define SWITCH_MODE            FOLLOW_INV            // [SwitchMode] TOGGLE, FOLLOW, FOLLOW_INV, PUSHBUTTON, PUSHBUTTON_INV, PUSHBUTTONHOLD, PUSHBUTTONHOLD_INV, PUSHBUTTON_TOGGLE, TOGGLEMULTI, FOLLOWMULTI, FOLLOWMULTI_INV (the wall switch state)
#define APP_DISABLE_POWERCYCLE true             // [SetOption65] Disable fast power cycle detection for device reset

 #define ETH_TYPE          0                    // [EthType] 0 = LAN8720, 1 = TLK110/IP101, 2 = RTL8201, 3 = DP83848, 4 = DM9051, 5 = KSZ8081, 6 = KSZ8041, 7 = JL1101, 8 = W5500, 9 = KSZ8851
 #define ETH_ADDRESS       1                    // [EthAddress] 0 = PHY0 .. 31 = PHY31
 #define ETH_CLKMODE       0                    // [EthClockMode] 0 = ETH_CLOCK_GPIO0_IN, 1 = ETH_CLOCK_GPIO0_OUT, 2 = ETH_CLOCK_GPIO16_OUT, 3 = ETH_CLOCK_GPIO17_OUT

#define USER_BACKLOG "Module 0;GPIO2 160;GPIO4 161;GPIO12 162;GPIO14 163;GPIO15 164;GPIO35 165;GPIO36 166;GPIO39 167;switchtext1 DW1;switchtext2 DW2;switchtext3 DW3;switchtext4 DW4;switchtext5 DW5;switchtext6 DW6;switchtext7 DW7;switchtext8 DW8"}
#endif  // _USER_CONFIG_OVERRIDE_H_
```

**Reply by sfromis** [reply]
Indeed semicolon as delimiter, just like with the `Backlog` console command I already mentioned.

But do not include the `Module 0` command, you as you are already setting the module to activate the template when you have `#define MODULE USER_MODULE`

And I'd really recommend not using those GPIO commands, but have the settings in a combined template with all pins. You are not "supposed" to keep the template to the default initial setup. Using "Configure Template", you can change them easily, and then use the saved template after that. No `GPIO` commands in the template either.

**Reply by dali-dbr** [reply]
Thank you.

Now I understand.

Use the Configure Template
![image](https://github.com/user-attachments/assets/20c0c046-08cc-434e-b487-2a294a8556b4)

Set the GPIOS
![image](https://github.com/user-attachments/assets/015663c0-ee2d-4a6a-be65-4dee3473580e)

export the template in the Console using the command **template**

Result:
{"NAME":"WT32_ETH01","GPIO":[1,0,160,1,161,1,0,0,162,0,163,164,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,0,165,166,0,0,167],"FLAG":0,"BASE":1}

**Reply by dali-dbr** [reply]
This is updated user_config_override.h

```
#define PROJECT                "Dali_DW"         // PROJECT is used as the default topic delimiter

// If not selected the default will be SONOFF_BASIC
#ifdef MODULE
#undef MODULE
#endif
#define MODULE                 USER_MODULE   // [Module] Select default model (the list is kModuleNiceList() in file tasmota_template.h) USER_MODULE is the TEMPLATE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE        WT32_ETH01   // to Select the default model as FALLBACK when the user does a RESET 1 

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE          "{"NAME":"WT32_ETH01","GPIO":[0,32,160,0,161,0,0,0,162,0,163,164,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,0,0,0,165,166,0,0,167],"FLAG":0,"BASE":1}" // [Template] Set JSON template


#define SAVE_DATA              1                 // [SaveData] Save changed parameters to Flash (0 = disable, 1 - 3600 seconds)
#define SAVE_STATE             true              // [SetOption0] Save changed power state to Flash (false = disable, true = enable)
#define BOOT_LOOP_OFFSET       1                 // [SetOption36] Number of boot loops before starting restoring defaults (0 = disable, 1..200 = boot loops offset)


#define MQTT_SWITCHES          true             // [SetOption114] Detach switches from relays and send MQTT messages instead
#define SWITCH_MODE            FOLLOW_INV            // [SwitchMode] TOGGLE, FOLLOW, FOLLOW_INV, PUSHBUTTON, PUSHBUTTON_INV, PUSHBUTTONHOLD, PUSHBUTTONHOLD_INV, PUSHBUTTON_TOGGLE, TOGGLEMULTI, FOLLOWMULTI, FOLLOWMULTI_INV (the wall switch state)
#define APP_DISABLE_POWERCYCLE true             // [SetOption65] Disable fast power cycle detection for device reset

 #define ETH_TYPE          0                    // [EthType] 0 = LAN8720, 1 = TLK110/IP101, 2 = RTL8201, 3 = DP83848, 4 = DM9051, 5 = KSZ8081, 6 = KSZ8041, 7 = JL1101, 8 = W5500, 9 = KSZ8851
 #define ETH_ADDRESS       1                    // [EthAddress] 0 = PHY0 .. 31 = PHY31
 #define ETH_CLKMODE       0                    // [EthClockMode] 0 = ETH_CLOCK_GPIO0_IN, 1 = ETH_CLOCK_GPIO0_OUT, 2 = ETH_CLOCK_GPIO16_OUT, 3 = ETH_CLOCK_GPIO17_OUT

#define USER_BACKLOG "switchtext1 DW1;switchtext2 DW2;switchtext3 DW3;switchtext4 DW4;switchtext5 DW5;switchtext6 DW6;switchtext7 DW7;switchtext8 DW8"

#endif  // _USER_CONFIG_OVERRIDE_H_
```

**Comment by dali-dbr** [comment]
Hi.

Today I managed to build a custom firmware using PlatformIO and was able to upload it via [online tasmota installer].(https://tasmota.github.io/install/).  Don't know yet how to upload using Platformio.

Since this is very new to me this is a huge step for me ;)

I'm using Platformio tasmota32  build command and uploading firmware.factory.bin to the device. I hope that's correct.

![image](https://github.com/user-attachments/assets/83d93760-ce7e-430e-a7b6-d5e440d1533f)


I got the Ethernet port and user template working from the first boot. 

however,

#define USER_BACKLOG "switchtext1 DW1;switchtext2 DW2;switchtext3 DW3;switchtext4 DW4;switchtext5 DW5;switchtext6 DW6;switchtext7 DW7;switchtext8 DW8"

this will rename only the first switch. The rest (after semicolon delimiter) are not renamed.
![image](https://github.com/user-attachments/assets/18f91bfa-0707-4fdc-b807-32d9ae4a0cf6)

bellow is my user_config_override.h

```
#define PROJECT                "Dali_DW"         // PROJECT is used as the default topic delimiter

// If not selected the default will be SONOFF_BASIC
#ifdef MODULE
#undef MODULE
#endif
#define MODULE                 USER_MODULE   // [Module] Select default model (the list is kModuleNiceList() in file tasmota_template.h) USER_MODULE is the TEMPLATE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE        USER_MODULE   // to Select the default model as FALLBACK when the user does a RESET 1 

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE          "{\"NAME\":\"WT32_ETH01\",\"GPIO\":[1,0,160,1,161,1,0,0,162,0,163,164,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,0,165,166,0,0,167],\"FLAG\":0,\"BASE\":1}" // [Template] Set JSON template


#define SAVE_DATA              1                 // [SaveData] Save changed parameters to Flash (0 = disable, 1 - 3600 seconds)
#define SAVE_STATE             true              // [SetOption0] Save changed power state to Flash (false = disable, true = enable)
#define BOOT_LOOP_OFFSET       1                 // [SetOption36] Number of boot loops before starting restoring defaults (0 = disable, 1..200 = boot loops offset)


#define MQTT_SWITCHES          true             // [SetOption114] Detach switches from relays and send MQTT messages instead
#define SWITCH_MODE            FOLLOW_INV            // [SwitchMode] TOGGLE, FOLLOW, FOLLOW_INV, PUSHBUTTON, PUSHBUTTON_INV, PUSHBUTTONHOLD, PUSHBUTTONHOLD_INV, PUSHBUTTON_TOGGLE, TOGGLEMULTI, FOLLOWMULTI, FOLLOWMULTI_INV (the wall switch state)
#define APP_DISABLE_POWERCYCLE true             // [SetOption65] Disable fast power cycle detection for device reset

#define USE_WT32_ETH01                         // Add support for Wireless-Tag WT32-ETH01
#define ETH_TYPE          0                    // [EthType] 0 = LAN8720, 1 = TLK110/IP101, 2 = RTL8201, 3 = DP83848, 4 = DM9051, 5 = KSZ8081, 6 = KSZ8041, 7 = JL1101, 8 = W5500, 9 = KSZ8851
#define ETH_ADDRESS       1                    // [EthAddress] 0 = PHY0 .. 31 = PHY31
#define ETH_CLKMODE       0                    // [EthClockMode] 0 = ETH_CLOCK_GPIO0_IN, 1 = ETH_CLOCK_GPIO0_OUT, 2 = ETH_CLOCK_GPIO16_OUT, 3 = ETH_CLOCK_GPIO17_OUT

#define USER_BACKLOG "switchtext1 DW1;switchtext2 DW2;switchtext3 DW3;switchtext4 DW4;switchtext5 DW5;switchtext6 DW6;switchtext7 DW7;switchtext8 DW8"

#endif  // _USER_CONFIG_OVERRIDE_H_

```

Thank you for your help!

Br

**Comment by UBWH** [comment]
I read above that these can be set at build time
```
#ifdef MODULE
#undef MODULE
#endif
#define MODULE                 USER_MODULE   // [Module] Select default model (the list is kModuleNiceList() in file tasmota_template.h) USER_MODULE is the TEMPLATE

#ifdef FALLBACK_MODULE
#undef FALLBACK_MODULE
#endif
#define FALLBACK_MODULE        USER_MODULE   // to Select the default model as FALLBACK when the user does a RESET 1 

#ifdef USER_TEMPLATE
#undef USER_TEMPLATE
#endif
#define USER_TEMPLATE          "{\"NAME\":\"WT32_ETH01\",\"GPIO\":[1,1,1,1,1,1,0,0,1,0,1,1,3840,576,5600,0,0,0,0,5568,0,0,0,0,0,0,0,0,1,1,0,1,1,0,0,1],\"FLAG\":0,\"BASE\":1}" 
```

Great. Have done it myself.
But that is a pain as every time there is a new FW release, have to build a new binary to get latest features.

My question is: Can these setting be done instead a boot time - e.g. with one of the `autoexec` files?

[PS: Yes I noted the comment above not to set MODULE & FALLBACK_MODULE the same]

**Reply by sfromis** [reply]
Well, you do not need to set template/module at boot, and neither to build new binaries with every new release. These settings will be automatically persisted across boots and version upgrades.

Of course, you will have to do "something" if you make a "cold" erase/install instead of upgrading, which would mean setting up your autoexec, autoconf or tapp files do something. Sure, if you install some Berry code ways, it would be simple to check if a template has been configured, and take action if not. And the autoconf feature is designed for having such a package taking install action first time included in boot, as long as you "somehow" ensure that the file is installed after a "cold" erase/install.

Still, if you do not use erase/flash, but "only" a full reset, what you have in the file system should survive and be able to take actions upon boot if detecting a setup need.

**Comment by UBWH** [comment]
@sfromis: Thanks

My objective is to have these survive a `RESET 1`

1. MODULE
2. FALLBACK_MODULE
3. USER _TEMPLATE

I know the UFS file system survives a `RESET 1` (Good)

So I could put commands in `autoexec.bat`  (or `autoexec.be`) as below.

1. `MODULE 0`   See comment below
2. FALLBACK_MODULE = Not really needed if MODULE is set in autoexec.bat
3. `TEMPLATE <template_string>`

Using `MODULE 0` does a reboot - so dumb idea to put that in autoexec.xxx (=> Boot loop)
But .... what about this logic in `autoexec.be`

```
Pseudocode

DesiredTemplate = 'xxxx'
if(CurrentTemplate != DesiredTemplate) {
  Template  = DesiredTemplate             <------ does not cause a reboot
}

DesiredModule =0
if(CurrentModule != DesiredModule) {
  Module = DesiredModule                     <--------- Will reboot, but will not execute this code next boot up
}
```
Is my logic sound?

**Reply by sfromis** [reply]
Yeah, as I wrote:
> ... and be able to take actions upon boot if detecting a setup need.

However, your pseudo-code does not have anything operationally usable.
```
if tasmota.cmd('Status',true)['Status']['Module'] == 1
  tasmota.cmd(f'Template {DesiredTemplate}')
  tasmota.cmd('Module 0')
end
```
Having a `DesiredModule` is typically not very useful, as a template needs `Module 0`, and with ESP32, the old device type specific `Module` concept is left in the dust.

**Reply by UBWH** [reply]
Built the latest DEV version today:
Tasmota Version 15.0.1.2(tasmota32)-3_1_3(2025-07-28T09:26:03)

The `MODULE 0` command reboots the device but does NOT change the module

```
10:45:43.028 CMD: template
10:45:43.030 RSL: RESULT = {"NAME":"Lilygo-T3S3","GPIO":[32,1,1,672,1,736,704,10656,10688,1,1,1,1,1,1,1,1,608,640,1,1,1,10784,10720,1,1,0,1,1,0,1,0,6210,1,1,1,1,0],"FLAG":0,"BASE":1}
10:45:50.984 CMD: module 
10:45:50.985 RSL: RESULT = {"Module":{"1":"ESP32S3"}}
10:48:17.775 CMD: module 0
10:48:17.776 RSL: RESULT = {"Module":{"0":"Lilygo-T3S3"}}
```
Tasmota reboots as expected
```
...
10:49:03.488 CMD: module
10:49:03.489 RSL: RESULT = {"Module":{"1":"ESP32S3"}}
10:49:11.040 CMD: template
10:49:11.042 RSL: RESULT = {"NAME":"Lilygo-T3S3","GPIO":[32,1,1,672,1,736,704,10656,10688,1,1,1,1,1,1,1,1,608,640,1,1,1,10784,10720,1,1,0,1,1,0,1,0,6210,1,1,1,1,0],"FLAG":0,"BASE":1}
```

When I re-do same test with `Version":"15.0.1.2(48c482d-tasmota32)` (Nightly build 2 days(?) ago)  ... works as expected
```
10:53:32.976 CMD: template
10:53:32.980 MQT: stat/LWG-T3S3_5390B0/RESULT = {"NAME":"LilygoT3S3-SX126x","GPIO":[32,1,1,672,1,736,704,10656,10688,1,1,1,1,1,1,1,1,608,640,1,1,1,10784,10720,1,1,0,1,1,0,1,0,6210,1,1,1,1,0],"FLAG":0,"BASE":1}
10:53:37.677 CMD: module
10:53:37.680 MQT: stat/LWG-T3S3_5390B0/RESULT = {"Module":{"1":"ESP32S3"}}
10:53:52.533 CMD: module 0
10:53:52.536 MQT: stat/LWG-T3S3_5390B0/RESULT = {"Module":{"0":"LilygoT3S3-SX126x"}}
```
Reboot
```
10:54:42.031 CMD: module
10:54:42.034 MQT: stat/LWG-T3S3_5390B0/RESULT = {"Module":{"0":"LilygoT3S3-SX126x"}}
10:54:46.446 CMD: template
10:54:46.450 MQT: stat/LWG-T3S3_5390B0/RESULT = {"NAME":"LilygoT3S3-SX126x","GPIO":[32,1,1,672,1,736,704,10656,10688,1,1,1,1,1,1,1,1,608,640,1,1,1,10784,10720,1,1,0,1,1,0,1,0,6210,1,1,1,1,0],"FLAG":0,"BASE":1}
```

Any ideas?

**Reply by UBWH** [reply]
UPDATE: Just noticed this on the newly compiled version:
```
INFO3 = {"Info3":{"RestartReason":{"Exception":28,"Reason":"LoadProhibited","EPC":"4200395d","EXCVADDR":"00000000","CallChain":["4200395a","42029432","4206d429","4200ee7c","4200561c","42014e04","4200a011","420a4692"]},"BootCount":13}}
```
That is a crash

**Reply by UBWH** [reply]
The 'good' FW
```
"Info3":{"RestartReason":"Software reset CPU","BootCount":122}}
```

**Reply by sfromis** [reply]
Well, Tasmota does try to recover from boot loops by "scaling back" configs until able to boot without quickly crashing *again*. With no details of what you did in your build, no basis for guesses about why it crashes.

**Reply by UBWH** [reply]
Code changes by Theo fixed this :-)
